ogr/ogrsf_frmts/miramon/ogrmiramonlayer.cpp

```diff
@@ -0,0 +1,2613 @@
+                                            "r")) == nullptr)
+                    {
+                        CPLDebug("MiraMon", "File '%s' cannot be opened.",
+                                 phMiraMonLayer->pMMBDXP->szFileName);
+                        bValidFile = false;
+                        return;
+                    }
+
+                    // First time we open the extended DBF we create an index
+                    // to fastly find all non geometrical features.
+                    phMiraMonLayer->pMultRecordIndex = MMCreateExtendedDBFIndex(
+                        phMiraMonLayer->pMMBDXP->pfDataBase,
+                        phMiraMonLayer->pMMBDXP->nRecords,
+                        phMiraMonLayer->pMMBDXP->FirstRecordOffset,
+                        phMiraMonLayer->pMMBDXP->BytesPerRecord,
+                        phMiraMonLayer->pMMBDXP
+                            ->pField[phMiraMonLayer->pMMBDXP->IdGraficField]
+                            .AcumulatedBytes,
+                        phMiraMonLayer->pMMBDXP
+                            ->pField[phMiraMonLayer->pMMBDXP->IdGraficField]
+                            .BytesPerField,
+                        &phMiraMonLayer->isListField, &phMiraMonLayer->nMaxN);
+
+                    // Creation of maximum number needed for processing
+                    // multiple records
+                    if (phMiraMonLayer->pMultRecordIndex)
+                    {
+                        papszValues = static_cast<char **>(
+                            CPLCalloc((size_t)phMiraMonLayer->nMaxN + 1,
+                                      sizeof(*papszValues)));
+
+                        padfValues = static_cast<double *>(
+                            CPLCalloc((size_t)phMiraMonLayer->nMaxN,
+                                      sizeof(*padfValues)));
+                    }
+
+                    phMiraMonLayer->iMultiRecord =
+                        -2;  // No option iMultiRecord
+                    const char *szMultiRecord =
+                        CSLFetchNameValue(papszOpenOptions, "iMultiRecord");
+                    if (phMiraMonLayer->isListField && szMultiRecord)
+                    {
+                        if (!strcasecmp(szMultiRecord, "Last"))
+                            phMiraMonLayer->iMultiRecord = -1;
+                        else if (!strcasecmp(szMultiRecord, "JSON"))
+                            phMiraMonLayer->iMultiRecord = -3;
+                        else
+                            phMiraMonLayer->iMultiRecord = atoi(szMultiRecord);
+                    }
+                }
+
+                for (MM_EXT_DBF_N_FIELDS nIField = 0;
+                     nIField < phMiraMonLayer->pMMBDXP->nFields; nIField++)
+                {
+                    OGRFieldDefn oField("", OFTString);
+                    oField.SetName(
+                        phMiraMonLayer->pMMBDXP->pField[nIField].FieldName);
+
+                    oField.SetAlternativeName(
```

Comment:
Testing with autotest/ogr/data/miramon/Arcs/EmptyArcs/Empty_ARC.arc, I see alternative name isn't necessary encoded in UTF-8. This may be somewhat problematic. If it can't be recoded to UTF-8 it would be best not to set the alternative name

Reply:
Sorry, I don't understand what you mean with UTF-8 here.

Reply:
well, the value set as the alternative name with file autotest/ogr/data/miramon/Arcs/EmptyArcs/Empty_ARC.arc is *not* encoded in UTF-8, but in the native encoding of the file (LATIN-1 I assume). It is preferable to not use non-UTF-8 values when setting strings with the OGR API. So if you know the encoding of the value, it would be best to use CPLRecode() to convert it to UTF-8 before setting it with SetAlternativeName()

Reply:
Ah, okay! Sure! I forgot to do that. Usually, I have done that in other cases. Hopefully, I don't forget anyone else.

Reply:
I didn't forget it. In fact I was recoding twice. Inside MMReturnValueFromSectionINIFile() after CPLReadLine2L() and at the return of the function (on // MM_DEF_LANGUAGE) in mm_gdal_functions.c.

Once this is solved it should work. But I observe that in cmd line things are not well printed but I think that is a problem of how it's printed... it's possible?

![image](https://github.com/AbelPau/gdal/assets/92721356/26c400fa-5107-41b5-9229-94e2c00b0cc9)

Reply:
Ah you're still using Windows ;-) (Windows is just for end users, not developers, if you want my opinion). Well Windows console doesn't know how to display UTF-8 properly by default.  Cf https://stackoverflow.com/questions/57131654/using-utf-8-encoding-chcp-65001-in-command-prompt-windows-powershell-window as a way of forcing it to use UTF-8. Otherwise redirect the output to a file "ogrinfo bla.arc > out.txt" and open it with a UTF-8 capable editor (notepad++, etc)

Reply:
Yes, we have been using windows for 30 years. To change now is a little bit difficult :)